lib/libxdp/libxdp.c

```diff
@@ -1433,9 +1433,11 @@ struct xdp_program *xdp_program__from_id(__u32 id)
 	}
 
 	prog = xdp_program__from_fd(fd);
+	// duplicated fd already in prog, close original
+	if(fd > 0)
+		close(fd);
```

Comment:
There's a check right above this for the validity of fd, so the if statement here is redundant. The same goes for all the other places you're doing changing below :)

Reply:
Yeah, you're right, I went too far, but thanks to that I realized one problem with fd/hw_fd handling